src/vcpkg/postbuildlint.cpp

```diff
@@ -577,16 +576,35 @@ namespace vcpkg::PostBuildLint
             {
                 binaries_with_invalid_architecture.push_back({file, actual_architecture});
             }
-        }
+        };
+#elif defined(__APPLE__)
+        const auto requested_arch = expected_architecture == "x64" ? "x86_64" : expected_architecture;
+        const auto check_lib = [&](const Path& file) {
+            auto cmd_line = Command("lipo").string_arg("-archs").path_arg(file);
+            ExitCodeAndOutput ec_data = cmd_execute_and_capture_output(cmd_line);
+            Checks::check_exit(VCPKG_LINE_INFO,
```

Comment:
Since we have downgraded all checks to optional, it no longer makes sense to hard crash if we fail to run `lipo` or `dumpbin`.